client/state/account-recovery/reset/reducer.js

```diff
@@ -35,6 +37,18 @@ const resetOptions = combineReducers( {
 	items,
 } );
 
+const userDataFieldReducer = ( expectedField ) => createReducer( '', {
+	[ ACCOUNT_RECOVERY_RESET_UPDATE_USER_DATA ]: ( state, action ) => null != action[ expectedField ] ? action[ expectedField ] : state,
+} );
+
+const userData = combineReducers( {
+	user: userDataFieldReducer( 'user' ),
+	firstName: userDataFieldReducer( 'firstName' ),
+	lastName: userDataFieldReducer( 'lastName' ),
+	url: userDataFieldReducer( 'url' ),
+} );
+
```

Comment:
Another way to write this without creating reducers for each property (and less tests):

```js
import { pick } from 'lodash';

const validProps = [ 'user', 'firstName', 'lastName', 'url' ];
const userData = createReducer( {}, {
	[ ACCOUNT_RECOVERY_RESET_UPDATE_USER_DATA ]: ( state, action ) =>
		{ ...state, ...pick( action, validProps ) }
} );
```